src/shogun/preprocessor/PruneVarSubMean.cpp

```diff
@@ -115,41 +99,33 @@ void CPruneVarSubMean::cleanup()
 	m_initialized = false;
 }
 
-/// apply preproc on feature matrix
-/// result in feature matrix
-/// return pointer to feature_matrix, i.e. f->get_feature_matrix();
-SGMatrix<float64_t> CPruneVarSubMean::apply_to_feature_matrix(CFeatures* features)
+SGMatrix<float64_t>
+CPruneVarSubMean::apply_to_matrix(SGMatrix<float64_t> matrix)
 {
 	ASSERT(m_initialized)
 
-	int32_t num_vectors=0;
-	int32_t num_features=0;
-	float64_t* m=((CDenseFeatures<float64_t>*) features)->get_feature_matrix(num_features, num_vectors);
+	int32_t num_vectors = matrix.num_cols;
 
-	SG_INFO("get Feature matrix: %ix%i\n", num_vectors, num_features)
-	SG_INFO("Preprocessing feature matrix\n")
-	for (int32_t vec=0; vec<num_vectors; vec++)
+	for (auto i : range(num_vectors))
 	{
-		float64_t* v_src=&m[num_features*vec];
-		float64_t* v_dst=&m[m_num_idx*vec];
+		auto v_src = matrix.get_column(i);
+		auto v_dst = matrix.get_column(i);
```

Comment:
m_num_idx*vec == num_features*vec ?

as now `v_src` and `v_dst` points to the very same memory region, before the patch idk. just checking.

Reply:
my mistake, thanks